subprojects/language-java/src/main/java/org/gradle/api/tasks/compile/CompileOptions.java

```diff
@@ -50,7 +51,7 @@
     private static final long serialVersionUID = 0;
 
     private static final ImmutableSet<String> EXCLUDE_FROM_ANT_PROPERTIES =
-            ImmutableSet.of("debugOptions", "forkOptions", "compilerArgs", "incremental");
+            ImmutableSet.of("debugOptions", "forkOptions", "compilerArgs", "incremental", "allCompilerArgs", "compilerArgumentProviders");
```

Comment:
Will our command-line argument providers work with Ant?

Reply:
Nope, but we still have some code to convert `CompileOptions` to some ant properties. I think this is not used anywhere (we don't have an Ant based compiler anymore), but the exclusion code is still there. Actual tests break if we don't exclude the new methods from this list.

I don't know if we still need to support [optionMap](https://github.com/gradle/gradle/blob/c62dcb279428c68d0ad6162ff25e7225d2e6d9ec/subprojects/language-jvm/src/main/java/org/gradle/api/tasks/compile/AbstractOptions.java#L45), which seems to be the place where we use the exclude list.

Reply:
Let's take a look later if we can remove `optionMap()` as a thing.